internal/resources/cloudprovider/data_source_aws_cloudwatch_scrape_job_test.go

```diff
@@ -57,19 +64,41 @@ func TestAccDataSourceAWSCloudWatchScrapeJob(t *testing.T) {
 					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "custom_namespace.0.scrape_interval_seconds", fmt.Sprintf("%d", testAWSCloudWatchScrapeJobData.CustomNamespaces[0].ScrapeIntervalSeconds)),
 				),
 			},
+			// update to remove regions_subset_override so that the account's regions are used instead
+			{
+				Config: awsCloudWatchScrapeJobResourceData(testCfg.stackID,
+					jobName,
+					false,
+					testCfg.accountID,
+					"",
+					servicesString(testAWSCloudWatchScrapeJobData.Services),
+					customNamespacesString(testAWSCloudWatchScrapeJobData.CustomNamespaces),
+				) + awsCloudWatchScrapeJobDataSourceData(testCfg.stackID, jobName),
+				Check: resource.ComposeTestCheckFunc(
+					checkAWSCloudWatchScrapeJobResourceExists("grafana_cloud_provider_aws_cloudwatch_scrape_job.test", testCfg.stackID, &gotJob),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "stack_id", testCfg.stackID),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "name", jobName),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "regions.#", fmt.Sprintf("%d", len(testAWSCloudWatchScrapeJobData.Regions))),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "regions.0", testAWSCloudWatchScrapeJobData.Regions[0]),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "regions.1", testAWSCloudWatchScrapeJobData.Regions[1]),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "regions.2", testAWSCloudWatchScrapeJobData.Regions[2]),
+					resource.TestCheckResourceAttr("data.grafana_cloud_provider_aws_cloudwatch_scrape_job.test", "regions_subset_override_used", "false"),
+				),
+			},
 		},
+		CheckDestroy: checkAWSCloudWatchScrapeJobResourceDestroy(testCfg.stackID, &gotJob),
 	})
 }
 
-func awsCloudWatchScrapeJobDataSourceData() string {
+func awsCloudWatchScrapeJobDataSourceData(stackID string, jobName string) string {
 	data := fmt.Sprintf(`
 data "grafana_cloud_provider_aws_cloudwatch_scrape_job" "test" {
 	stack_id = "%[1]s"
-	name = "%[2]s"
+	name = grafana_cloud_provider_aws_cloudwatch_scrape_job.test.name
```

Comment:
shouldn't this be the `jobName` argument?

Reply:
IMO no, because we pass `jobName` into the scrape job resource to create, then reference that resource in this data source block, which makes Terraform depend on the resource to be created before doing the lookup.

Reply:
Imagine it's answer with same as below